### PR TITLE
feature/es6 support

### PR DESCRIPTION
**Pull request by viqueen**
This PR introduces es6 support to the repo.

**additions**

```
npm run test:watch
# replacing the current npm test

npm run lint          
# have a look at .eslintrc if you think the existing rules are enough

npm run lint:watch     
# lint in watch mode

npm run compile        
# important!! needs to be run before release, note how main script points to dist instead of src
```

**TODO** 
- update readme
- restore check-coverage values and extend test coverage

**nice to do**
- check for new dependency versions

```
npm install salita -g
salita .
```
- check for licenses  

```
npm install license-checker -g
license-checker .
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,13 +35,21 @@
...
-    "cz-conventional-changelog": "1.2.0",
-    "mocha": "3.1.2",
     "codecov.io": "0.1.6",
+    "commitizen": "2.8.6",
+    "cz-conventional-changelog": "1.2.0",
+    "eslint": "^3.8.1",
+    "eslint-watch": "^2.1.14",
     "ghooks": "1.3.2",
-    "istanbul": "0.4.5",
+    "istanbul": "^1.0.0-alpha.2",
```
Comment (viqueen) [inline package.json:51]: seems to work with babel

File: package.json [hunk 2]
```diff
@@ -49,6 +61,7 @@
     }
   },
   "dependencies": {
-    "text2token": "1.0.3"
+    "text2token": "1.0.3",
+    "underscore": "^1.8.3"
```
Comment (viqueen) [inline package.json:65]: to avoid altering Array.prototype methods

**Comment by accraze** [comment]
Thanks @viqueen!! This is awesome :+1: 

I'm looking at the TODOS section, what needs to be added to the readme?

**Comment by viqueen** [comment]
the Readme only needs the updated dev scripts mentioned in the pr description under additions :D , I got dragged into real life work and just didn't have time to wrap up the last PR details

**Comment by accraze** [comment]
Cool! no worries, I can add those updated dev scripts to the README. Thanks again for your contributions! :+1:

**Pull request merged**